### DISCUSSION TITLE
Nightscout-FreeDNS-Bad Gateway

### DISCUSSION THREAD

**Discussion by Loax28** [NightscoutFoundation/xDrip#3377] [Bug reports]
Hey,

hope this is what you meant @Navid200 

I tried to update to NS15. So I tried to update the platform, then I got the red comment in status: start up dependence on freeDNS + FreeDNS ID and pass.
Therefore I did in the terminal phase 2 in Google cloud setup.
But now I cannot access to my nightscout (Bad gateway).
In installation I first used the first way where I keep my certificate and then the second where I redo it. But I didn't to the steps after if that is needed.
I didn't do the first phase.
Thank you for any help.
![fdece5ce-21ee-40be-b45b-118e44487e76](https://github.com/NightscoutFoundation/xDrip/assets/133046189/ef306717-3857-4e03-b469-75ffec9b273a)

This was also recommended:
![image](https://github.com/NightscoutFoundation/xDrip/assets/133046189/9a9d78ee-8cb6-4c67-9273-dccc377a8201)

**Comment by Navid200** [comment]
Since you have posted two screenshots, for clarity, I need to know what you mean by "This was also recommended".

Who recommended any of that?

**Reply by Loax28** [reply]
https://github.com/NightscoutFoundation/xDrip/discussions/2788#discussioncomment-5659798

**Reply by Loax28** [reply]
and the 2nd is what he recommended

**Comment by Navid200** [comment]
Restart the server: https://navid200.github.io/xDrip/docs/Nightscout/Restart.html

Then, wait 30 seconds and then, use SSH again to reconnect.
What does the status page show then?

**Comment by Loax28** [comment]
I tried to reboot many times. It doesn't work
the upper status is the actual one. before it showed the problem with FreeDNS. Thanks for your help btw.

**Reply by Navid200** [reply]
In that case, run phase 1.

**Comment by Loax28** [comment]
hope it was okay, that I took the existing certificate in phase 2, after I did the first.

**Reply by Navid200** [reply]
I asked you to run phase 1.

**Reply by Loax28** [reply]
![image](https://github.com/NightscoutFoundation/xDrip/assets/133046189/df235e04-187c-4fc2-bed0-2779d21c5d31)
this was the status after phase 1.
status after phase 2 is the same

**Comment by Loax28** [comment]
It is working now btw😅

**Reply by Navid200** [reply]
Phase 2 attempts to creates a new certificate.  And if a certificate has been already created recently, it will bring up a warning that could confuse some.  There is really no need to run phase 2 unless it is needed.  
Anyway, I am glad it is working now.

**Comment by Heimdallr01** [comment]
I was having a similar problem with the Bad Gateway. I restarted several times, found this here, ran phase 1, restarted and nothing changed. It was still showing “Missing node_modules”. I restarted, ran phase 1, restarted and now I only get this console window

![image](https://github.com/NightscoutFoundation/xDrip/assets/31692758/9582f7c5-1042-4ed5-9033-ac33dfa6cb4b)

**Reply by Navid200** [reply]
What do you see if you type and enter the following?

`df -h`

**Reply by Heimdallr01** [reply]
![image](https://github.com/NightscoutFoundation/xDrip/assets/31692758/daee7d54-30cc-4ffa-a697-ec8f390dd280)

**Reply by Navid200** [reply]
Type the following for another reboot.  Wait for it to come back up.  Then, use SSH again to connect.  Does the menu come up?

`sudo reboot`

**Reply by Heimdallr01** [reply]
No, I get the console window I first posted

**Reply by Navid200** [reply]
Type and enter the following to do a bootstrap.  

`curl https://raw.githubusercontent.com/jamorham/nightscout-vps/vps-1/bootstrap.sh | bash`

**Reply by Heimdallr01** [reply]
Alright, now I can access the menu again. This is the status:

![image](https://github.com/NightscoutFoundation/xDrip/assets/31692758/a760f495-bd6d-4a42-992f-326642303f78)

I now keep getting a "The connection has timed out" though.

**Reply by Navid200** [reply]
You get the connection has timed out where?
Can you post a screenshot as long as it does not show any confidential data?

**Reply by Heimdallr01** [reply]
When trying to open my Nightscout website via IP and if I use my URL from FreeDNS I get "502 Bad Gateway"

**Reply by Navid200** [reply]
Run phase 1.

**Reply by Heimdallr01** [reply]
It works again! Thanks a lot!

**Reply by Navid200** [reply]
You're welcome
